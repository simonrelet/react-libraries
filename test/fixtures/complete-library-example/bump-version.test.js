'use strict'

const fs = require('fs-extra')
const path = require('path')
const semver = require('semver')
const { getFixtureVersion, setFixtureVersion } = require('../../fixtureUtils')
const { callScriptInPackage } = require('../../scriptExecution')
const packageJSON = require('./package.json')

const CHANGELOG_PATH = path.join(__dirname, 'CHANGELOG.md')
// const README_PATH = path.join(__dirname, 'README.md')

describe(`${packageJSON.name} bump-version`, () => {
  let initialChangelogContent = ''
  let initialVersion = ''
  let version = ''

  beforeAll(async () => {
    initialVersion = await getFixtureVersion(__dirname)
    version = semver.inc(initialVersion, 'major')

    initialChangelogContent = await fs.readFile(CHANGELOG_PATH, 'utf8')
  })

  afterAll(async () => {
    await setFixtureVersion(__dirname, initialVersion)
    await fs.writeFile(CHANGELOG_PATH, initialChangelogContent)
  })

  it('should not fail', async () => {
    const result = await callScriptInPackage(__dirname, 'bump-version', [
      version,
    ])

    expect(result).toMatchSnapshot()
  })

  it('should bump the new version', async () => {
    const bumpedVersion = await getFixtureVersion(__dirname)
    expect(bumpedVersion).toEqual(version)
  })

  it('should update the Changelog', async () => {
    const changelogContent = await fs.readFile(CHANGELOG_PATH, 'utf8')
    expect(changelogContent).toMatch(new RegExp(`${version}`))
  })
})
