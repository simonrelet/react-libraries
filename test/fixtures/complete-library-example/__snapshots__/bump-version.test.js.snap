// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complete-library-example bump-version should not fail 1`] = `
Exit code: 0

--- Output ---
package.json -> package.json
package-lock.json -> package-lock.json
CHANGELOG.md -> CHANGELOG.md
README-template.md -> README.md
--------------
`;
